src/workloads/execution/ShardedGraphLookup.yml

```diff
@@ -0,0 +1,372 @@
+    # distributed across the shards.
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: &Database test
+    - OperationMetricsName: ShardLocalCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: test.Collection0
+        key: {shardKey: hashed}
+        numInitialChunks: &NumChunks 6
+    - OperationMetricsName: ShardForeignCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: test.Collection2
+        key: {shardKey: hashed}
+        numInitialChunks: *NumChunks
+    # Disable the balancer so that it can't skew results while the $lookups are running.
+    - OperationMetricsName: DisableBalancer
+      OperationName: AdminCommand
+      OperationCommand:
+        balancerStop: 1
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: LoadGraphLookupData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  # We want the size of 'Collection2' to be smaller than the other two collections for the queries 
+  # where 'maxDepth' is not specified, so we first populate all three collections with the number of 
+  # documents we want in 'Collection2' and then in the next phase add more to 'Collection0' and 
+  # 'Collection1'.
+  - Repeat: 1
+    BatchSize: 1000
+    Threads: 1
+    DocumentCount: *num_docs_base
+    Database: *Database
+    CollectionCount: 3 # Loader will populate 'Collection0', 'Collection1', and 'Collection2'.
+    Document:
+      a: {^RandomInt: {min: 1, max: 100}}
+      b: {^RandomInt: {min: 1, max: 100}}
+      c: {^RandomInt: {min: 1, max: 10}}
+  - Repeat: 1
+    BatchSize: 1000
+    Threads: 1
+    DocumentCount: *num_docs_added
+    Database: *Database
+    CollectionCount: 2 # Loader will add additional documents to 'Collection0' and 'Collection1'.
+    Document:
+      a: {^RandomInt: {min: 1, max: 100}}
+      b: {^RandomInt: {min: 1, max: 100}}
+      c: {^RandomInt: {min: 1, max: 10}}
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        fsync: 1
+  - *Nop
+
+- Name: RunGraphLookups
+  Type: RunCommand
+  Database: *Database
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *Database
+    Operations:
+    # The recursive targeted queries below work as follows: for each document X in Collection0 
+    # we find any document Y in Collection0 where X's 'c' field (startWith) is equal to Y's 'b'
+    # (connectToField). Then, we look at Y's 'a' (connectFromField) value and find any document Z in 
+    # Collection0 that has the same value for its 'b' (connectToField), and so on. 
+    # Because the collection is sharded by 'b' the $graphLookup is targeted towards 
+    # specific shard(s).
```

Comment:
Above when you're sharding collections on lines 35 and 41, you're sharding by `shardKey` - which isn't used again. Did you mean to shard by `b`? Otherwise these "targeted" queries will actually be untargeted since they'd also have to do a scatter-gather to all the shards.

Reply:
Yes I did mean to shard by `b`! Thanks for catching that.